### PR TITLE
POST /documents to upload a document without a move

### PR DESCRIPTION
**Pull request by ngw**
It is now possible to POST to /documents and create a Document without a move

### CODE REVIEW

File: app/controllers/api/v1/documents_controller.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def document_params
 
       def document_attributes
         document_params[:attributes].merge(
-          move: Move.find(params.dig(:move_id)),
+          move: Move.find_by(id: params.dig(:move_id)),
```
Comment (starswan) [inline app/controllers/api/v1/documents_controller.rb]: did you mean this change? It causes a very slight behavioural change

File: db/schema.rb [hunk 2]
```diff
@@ -139,6 +139,27 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "notifications", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
```
Comment (starswan) [inline db/schema.rb]: oops this has escaped from your other branch

File: db/schema.rb [hunk 3]
```diff
@@ -208,6 +229,20 @@
     t.integer "latest_nomis_booking_id"
   end
 
+  create_table "subscriptions", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
```
Comment (starswan) [inline db/schema.rb]: ditto

File: app/controllers/api/v1/documents_controller.rb [hunk 4]
```diff
@@ -34,7 +34,7 @@ def document_params
 
       def document_attributes
```
Comment (starswan) [inline app/controllers/api/v1/documents_controller.rb]: as we are re-using this controller for 2 different (but very similar) endpoints, I think an intention comment would be useful.

Comment (ngw) [reply app/controllers/api/v1/documents_controller.rb]: well, the intention is for the code to do exactly what it does... there's a ticket about this, the reasoning behind it is not entirely clear to me (and I don't think it's relevant)

**Pull request merged**